cli/tools/upgrade.rs

```diff
@@ -21,6 +21,9 @@ use std::path::PathBuf;
 use std::process::Command;
 use std::time::Duration;
 
+if env!("TARGET") == "aarch64-apple-darwin" {
+  bail!("Canary builds are not available for M1");
+}
```

Comment:
This should be moved into the `upgrade` function below as this won't compile.

Reply:
Thanks for your comment.
Moved to upgrade function.